source/13_envs/sokoban_zh.rst

```diff
@@ -0,0 +1,134 @@
+Sokoban
+~~~~~~~
+
+概述
+=======
+
+Sokoban是一种离散动作空间环境。在该游戏中，智能体需要在离散动作集合中选择动作，目标是将房间中所有的箱子推到指定位置。这里的Sokoban环境实现了房间的随机生成，避免深度神经网络因为预定义的房间而训练过拟合。
+
+.. image:: ./images/sokoban.gif
+   :align: center
+   :scale: 70%
+
+安装
+=======
+
+安装方法
+--------
+
+用户可以选择通过 pip 一键安装或 git clone 代码库后 pip 本地安装
+
+注：如果用户没有 root 权限，请在 install 命令后加上 ``--user``
+
+.. code:: shell
+
+   # Method1: Install Directly
+   pip install gym-sokoban
+   # Method2: Install with DI-engine requirements
+   git clone git@github.com:mpSchrader/gym-sokoban.git
+   cd gym-sokoban
+   pip install -e .
+
+验证安装
+--------
+
+安装完成后，运行如下 Python 程序，如果没有报错则证明安装成功。
+
+.. code:: python
+
+   import gym
+   import gym_sokoban
+   env = gym.make('Sokoban-v0')
+   obs = env.reset()
+   print(obs.shape) # (160, 160, 3)
+
+变换前的空间（原始环境）
+==========================
+
+
+观察空间
+--------
+
+-  实际的游戏画面，RGB 三通道图片，具体尺寸为\ ``(160, 160, 3)``\ (以 Sokoban-v0 为例)，数据类型为\ ``uint8``
+
+-  每个房间中包含五种主要元素：墙壁，地板，箱子，目标点，和玩家，在游戏画面中以不同颜色标识。当箱子和玩家与目标点重合时，颜色也会发生变化。
+
+动作空间
+--------
+
+-  游戏提供了9种环境交互动作，形成大小为 9 的离散动作空间，数据类型为\ ``int``\ ，需要传入 python 数值（或是 0 维 np 数组，例如动作 3 为\ ``np.array(3)``\ ）。动作在 0-8 中取值，具体的含义是：
+
+   -  0：No Operation
+
+   -  1：Push Up
+
+   -  2：Push Down
+
+   -  3：Push Left
+
+   -  4：Push Right
+
+   -  5：Move Up
+
+   -  6：Move Down
+
+   -  7：Move Left
+
+   -  8：Move Right
+
+-  其中 Move 表示仅移动，需要对应方向上的下一个格子没有箱子或墙。
+
+-  Push 表示移动相邻的箱子，需要箱子的下一格是空闲的，不能直接推动两个相邻的箱子。如果对应方向上的相邻格子没有箱子，则 Push 和 Move 效果一致。
+
+奖励空间
+--------
+
+-  游戏得分，一般是一个\ ``float``\ 数值，具体的数值如下：
+
+   -  行动一步：-0.1
+
+   -  将箱子推到目标点上：1.0
+
+   -  将箱子推离目标点：-1.0
+
+   -  将所有箱子推到目标点上：10.0
+
+
+终止条件
+----
+
+-  所有箱子都推到目标点上，或者最大步数用完，则当前环境 episode 结束
+
+
+内置环境
+----
+
+-  Sokoban 内置了9个环境，分别为 ``Sokoban-v0``, ``Sokoban-v1``， ``Sokoban-v2``， ``Sokoban-small-v0``， ``Sokoban-small-v1``， ``Sokoban-large-v0``， ``Sokoban-large-v1``， ``Sokoban-large-v2``， ``Sokoban-huge-v0``。环境仅有房间大小和箱子数量差异，房间内部环境均为随机生成。
```

Comment:
Does this env random generate different layouts for each episode? Maybe you can add some details